### ISSUE TITLE
Test-Path -Path $null should return $false

### ISSUE THREAD

**Issue by KevinMarquette** [PowerShell/PowerShell#8076]
Test-Path will throw an error when the path is `$null` (or an empty string) instead of returning `$false`.

The errors that we do get are parameter binding errors

I find myself often running into errors with `Test-Path` because of empty or `$null` strings. Then I end up making my validation more complex to account for it.

```
if ( Test-Path -Path $Path)
{
    # do something
}

# VS

if ( $null -ne $Path -and ( Test-Path -Path $Path) )
{
    # do something
}
```
It would be nice if it would just return `$false` (or offer a `-Quiet` parameter)

Steps to reproduce
------------------

```powershell
$path = Resolve-Path -Path '.\*\nothing'
Test-Path $path
```
```powershell
Resolve-Path -Path '.\*\nothing' | Test-Path
```

Expected behavior
-----------------

```none
False
```

Actual behavior
---------------

```none
Error: Test-Path : Cannot bind argument to parameter 'Path' because it is null.
```

Environment data
----------------

```powershell
> $PSVersionTable
PSVersion                      6.1.0
PSEdition                      Core
GitCommitId                    6.1.0
OS                             Microsoft Windows 10.0.16299
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by dachshund-digital** [comment]
Over 2 years and this has not bee addressed?  Really?

**Comment by vexx32** [comment]
The team's choice was that `$null` is considered entirely unusable and they preferred that case should error.

Personally I think it should return $false, but that's what they decided.

**Comment by pabrams** [comment]
I'm pretty sure it _does_ return false?  It only stops execution if you have a non-default `$ErrorActionPreference`, like 'Stop'.  This is a non-terminating error.

**Comment by vexx32** [comment]
No, it doesn't return anything.

PowerShell will however treat the 'nothing' value as "false" if you use it in a context that will coerce the missing value to a boolean, like an `if` statement.

**Comment by pabrams** [comment]
I'm having a hard time imagining the practical value of treating nulls (or, really, empty strings was what I was thinking of) that way...  
Does the path "snickerdoodle" exist?  No.
Does the path "C:\temp" exist? Yes.
Does the path "" exist? No. There is no file or folder named after the empty string.
and, I guess:
Does the path null exist?  What? What the heck is that? It's definitely not a string or a path...

Maybe I'm not being fair.

**Comment by p00j4** [comment]
Upvoting the fix for this. 
I also have a need to use `Test-Path` on a computed path at runtime and at times it comes as `null` and hence it failed in the main pipeline with error  Cannot bind argument to parameter 'Path' because it is null. And now i'm forced to add one more check for "null". 
![image](https://user-images.githubusercontent.com/6470509/162166821-ea094fc8-f7e1-417a-a1cc-a05763ed49d5.png)


`PSVersion                      5.1.22580.1000`

**Comment by vexx32** [comment]
@p00j4 this case is easier to handle in current versions of PS, but no fixes are being backported to 5.1 other than absolutely critical vulnerabilities at this stage. 5.1 is no longer being actively maintained. I'd suggest you install one of the newer versions and use that 🙂
